Fix staticcheck/golint findings

There seems to be an updated version of a tool I used to use to statically check my Go source code: https://github.com/dominikh/go-tools

Luckily, `pina-golada` is already pretty good in that respect, but a couple of findings were found. This is something we could fix:
> internal/golada/cmd/generate.go:42:19: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (S1007)
> pkg/files/directory.go:181:2: redundant return statement (S1023)
> pkg/files/directory.go:270:2: redundant return statement (S1023)
> pkg/files/files_test.go:66:10: should use buffer.String() instead of string(buffer.Bytes()) (S1030)

Furthermore, the [Go Report Card](https://goreportcard.com/report/github.com/homeport/pina-golada) gives us a couple of more linter findings:
> pina-golada/pkg/annotation/property_parser.go
> Line 66: warning: redundant if ...; err != nil check, just return error instead. (golint)
>
> pina-golada/pkg/annotation/csv_parser.go
> Line 65: warning: redundant if ...; err != nil check, just return error instead. (golint)
>
> pina-golada/pkg/annotation/label_parser.go
> Line 65: warning: redundant if ...; err != nil check, just return error instead. (golint)

--- a/internal/golada/cmd/generate.go
+++ b/internal/golada/cmd/generate.go
@@ -39,7 +39,7 @@ import (
 
 var (
 	// GoFileSelector is the regex to select go files
-	GoFileSelector = regexp.MustCompile("(.*)\\.go")
+	GoFileSelector = regexp.MustCompile(`(.*)\.go`)
 
 	parserType string
 )

--- a/pkg/annotation/csv_parser.go
+++ b/pkg/annotation/csv_parser.go
@@ -60,10 +60,7 @@ func (c *CsvParser) Parse(comment string, annotation Annotation) (e error) {
 	if e != nil {
 		return e
 	}
-	removedQuotes := strings.Replace(string(out), "\"", "", -1)
 
-	if err := yaml.Unmarshal([]byte(removedQuotes), annotation); err != nil {
-		return err
-	}
-	return nil
+	removedQuotes := strings.Replace(string(out), "\"", "", -1)
+	return yaml.Unmarshal([]byte(removedQuotes), annotation)
 }

--- a/pkg/annotation/label_parser.go
+++ b/pkg/annotation/label_parser.go
@@ -60,10 +60,7 @@ func (c *LabelParser) Parse(comment string, annotation Annotation) (e error) {
 	if e != nil {
 		return e
 	}
-	removedQuotes := strings.Replace(string(out), "\"", "", -1)
 
-	if err := yaml.Unmarshal([]byte(removedQuotes), annotation); err != nil {
-		return err
-	}
-	return nil
+	removedQuotes := strings.Replace(string(out), "\"", "", -1)
+	return yaml.Unmarshal([]byte(removedQuotes), annotation)
 }

--- a/pkg/annotation/property_parser.go
+++ b/pkg/annotation/property_parser.go
@@ -62,9 +62,5 @@ func (p *PropertyParser) Parse(comment string, annotation Annotation) (e error) 
 	}
 
 	removedQuotes := strings.Replace(string(out), "\"", "", -1)
-
-	if err := yaml.Unmarshal([]byte(removedQuotes), annotation); err != nil {
-		return err
-	}
-	return nil
+	return yaml.Unmarshal([]byte(removedQuotes), annotation)
 }

--- a/pkg/files/directory.go
+++ b/pkg/files/directory.go
@@ -178,7 +178,6 @@ func (m *memoryDirectory) DeleteFile(path paths.Path) {
 			m.files = append(m.files[:index], m.files[index+1:]...)
 		}
 	}
-	return
 }
 
 // Directories lists all the directories in the directory
@@ -267,7 +266,6 @@ func (m *memoryDirectory) DeleteDirectory(path paths.Path) {
 			m.dirs = append(m.dirs[:index], m.dirs[index+1:]...)
 		}
 	}
-	return
 }
 
 // Parent returns the parent directory

--- a/pkg/files/files_test.go
+++ b/pkg/files/files_test.go
@@ -63,7 +63,7 @@ var _ = Describe("should handle files properly", func() {
 		Expect(file).To(Not(BeNil()))
 
 		Expect(file.CopyContent(buffer)).To(BeNil())
-		Expect(string(buffer.Bytes())).To(BeEquivalentTo("test"))
+		Expect(buffer.String()).To(BeEquivalentTo("test"))
 	})
 
 	_ = It("should return the correct paths", func() {

